### DISCUSSION TITLE
Alpaqa as casadi plugin

### DISCUSSION THREAD

**Discussion by jgillis** [kul-optec/alpaqa#6] [General]
Hi!

I would like to create a plugin for casadi, that can also be shipped with casadi.
What branch/release should I base my work on?

Are there any drastic API updates in the pipeline that complicate matters?

**Comment by tttapa** [comment]
Hi Joris,

That would be great!

At the moment, the `main` branch is quite far behind, so I would recommend using the [`develop`](https://github.com/kul-optec/alpaqa/tree/develop) branch.  

The problem API defined by [`alpaqa::TypeErasedProblem`](https://kul-optec.github.io/alpaqa/develop/Doxygen/classalpaqa_1_1TypeErasedProblem.html) should be fairly stable. You can find a bit more information on the [Problem formulation](https://kul-optec.github.io/alpaqa/develop/Doxygen/page_problem_formulations.html) page.  
The optimal control-specific APIs are not stable yet (e.g. [`alpaqa::TypeErasedControlProblem`](https://kul-optec.github.io/alpaqa/develop/Doxygen/classalpaqa_1_1TypeErasedControlProblem.html)), and require a very recent compiler (I would recommend disabling them using `-DALPAQA_WITH_OCP=Off`).

I do plan on changing the names of the statistics returned by the solvers (e.g. [`PANOCStats`](https://kul-optec.github.io/alpaqa/develop/Doxygen/namespacealpaqa.html#structalpaqa_1_1PANOCStats)), but this shouldn't cause too many issues.

If you need any help, just let me know and we can discuss further (either online or in person).
